From 353a8a8e3d44471e521839f56d36cef6ecfaa205 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Thu, 30 Jun 2022 20:04:35 +0200
Subject: [PATCH] build: Update geocode-glib libsoup3 dependency

geocode-glib 3.26.3 was released today, with optional soup3 support.
When built against soup3, it generates a geocode-glib-2.0 pkg-config
file rather than geocode-glib-1.0.
Adjust our geocode-glib dependency based whether libgweather uses
libsoup2 or libsoup3.
---
 meson.build | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 00fc2ff0..3596dea0 100644
--- a/meson.build
+++ b/meson.build
@@ -15,8 +15,13 @@ gobject = dependency('gobject-2.0', version: '>= 2.58')
 gtk = dependency('gtk4', version: '>= 4.5')
 gsound = dependency('gsound', version: '>=0.98')
 gweather = dependency('gweather4')
+gweather_soup_version = gweather.get_pkgconfig_variable('soupapiversion')
+if gweather_soup_version == '3.0'
+  geocodeglib = dependency('geocode-glib-2.0', version: '>=1.0')
+else
+  geocodeglib = dependency('geocode-glib-1.0', version: '>=1.0')
+endif
 gnomedesktop = dependency('gnome-desktop-4')
-geocodeglib = dependency('geocode-glib-1.0', version: '>=1.0')
 libgeoclue = dependency('libgeoclue-2.0', version: '>=2.4')
 libadwaita = dependency ('libadwaita-1', version: '>=1.0')
 
-- 
GitLab

